libraries/AP_Logger/AP_Logger_Block.cpp

```diff
@@ -137,10 +137,7 @@ bool AP_Logger_Block::_WritePrioritisedBlock(const void *pBuffer, uint16_t size,
         return false;
     }
 
-    if (!write_sem.take(1)) {
-         _dropped++;
-        return false;
-    }
+    write_sem.take_blocking();
```

Comment:
just switch to WITH_SEMAPHORE() and remove the write_sem.give() calls

Reply:
Fixed